src/hb-ot-color-sbix-table.hh

```diff
@@ -215,12 +215,13 @@ struct sbix
       hb_blob_t *blob = reference_blob_for_glyph (glyph, x_ppem, y_ppem,
 						  HB_TAG ('p','n','g',' '),
 						  &x_offset, &y_offset);
-      if (hb_blob_get_length (blob) < sizeof (PNGHeader))
+
+      const PNGHeader &header = *blob->as<PNGHeader>();
+      if (header.width == 0 && header.width == 0)
```

Comment:
No!  I meant, put the return at the end of function.  The whole point is to remove error-checking "if" statements.

Reply:
Hehehe, I see

Reply:
It will cause uninitialized memory read of x_offset and y_offset (fortunately tests caught it)

Reply:
Humm. need to understand why.

Oh, get_length < sizeof.... yes, you are right.  You still need it.  The as<> only checks for zero-length.  Otherwise assumes it's sane data.  Yeah, put the size check back please.

Reply:
I'm sorry. :)